src/engine/SCons/Script/Main.py

```diff
@@ -1104,12 +1104,21 @@ def _main(parser):
         if graph:
             SCons.Spirent.GraphWriter().write(graph, fs.Top)
 
-        # Build the targets
-        nodes = _build_targets(fs, options, targets, target_top)
-        if not nodes:
-            revert_io()
-            print('Found nothing to build')
-            exit_status = 2
+        # If the scons mode is set to vulcan, call the vulcan builder
+        scons_mode = os.environ.get("SCONS_MODE")
+        if options.mode == "vulcan" or (scons_mode == "vulcan" and options.mode != "local"):
```

Comment:
I think this probably works, but may be more maintainable as something like this:

```
scons_mode = options.mode or os.environ.get("SCONS_MODE")
if scons_mode == "vulcan":
    ... do vulcan
else:
    ... # do local
```

This way